test/api/v3/unit/middlewares/cronMiddleware.js

```diff
@@ -0,0 +1,170 @@
+import * as Tasks from '../../../../../website/src/models/task';
+import analyticsService from '../../../../../website/src/libs/api-v3/analyticsService';
+import { v4 as generateUUID } from 'uuid';
+
+describe('cron middleware', () => {
+  let res, req, next;
+  let user;
+
+  beforeEach(() => {
+    res = generateRes();
+    req = generateReq();
+    next = generateNext();
+    user = new User({
+      auth: {
+        local: {
+          username: 'username',
+          lowerCaseUsername: 'username',
+          email: 'email@email.email',
+          salt: 'salt',
+          hashed_password: 'hashed_password', // eslint-disable-line camelcase
+        },
+      },
+    });
+
+    user._statsComputed = {
+      mp: 10,
+      maxMP: 100,
+    };
+
+    res.locals.user = user;
+    res.analytics = analyticsService;
+  });
+
+  it('calls next when days have not been missed', () => {
+    cronMiddleware(req, res, next);
+    expect(next).to.be.calledOnce;
+  });
+
+  it('should clear todos older than 30 days for free users', (done) => {
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+    let task = generateTodo(res.locals.user);
+    task.dateCompleted = moment(new Date()).subtract({days: 31});
+    task.completed = true;
+    task.save();
+
+    cronMiddleware(req, res, function callback () {
+      Tasks.Task.findOne({_id: task}, function (err, taskFound) {
+        expect(err).to.not.exist;
+        expect(taskFound).to.not.exist;
+        done();
+      });
+    });
+  });
+
+  it('should not clear todos older than 30 days for subscribed users', (done) => {
+    res.locals.user.purchased.plan.customerId = 'subscribedId';
+    res.locals.user.purchased.plan.dateUpdated = moment('012013', 'MMYYYY');
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+    let task = generateTodo(res.locals.user);
+    task.dateCompleted = moment(new Date()).subtract({days: 31});
+    task.completed = true;
+    task.save();
+
+    cronMiddleware(req, res, function callback () {
+      Tasks.Task.findOne({_id: task}, function (err, taskFound) {
+        expect(err).to.not.exist;
+        expect(taskFound).to.exist;
+        done();
+      });
+    });
+  });
+
+  it('should clear todos older than 90 days for subscribed users', (done) => {
+    res.locals.user.purchased.plan.customerId = 'subscribedId';
+    res.locals.user.purchased.plan.dateUpdated = moment('012013', 'MMYYYY');
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+
+    let task = generateTodo(res.locals.user);
+    task.dateCompleted = moment(new Date()).subtract({days: 91});
+    task.completed = true;
+    task.save();
+
+    cronMiddleware(req, res, function callback () {
+      Tasks.Task.findOne({_id: task}, function (err, taskFound) {
+        expect(err).to.not.exist;
+        expect(taskFound).to.not.exist;
+        done();
+      });
+    });
+  });
+
+  it('should call next is user was not modified after cron', (done) => {
```

Comment:
Ahh, never mind, you swapped out passing in next _for_ a callback, so it's called implicitly by calling done(); Carry on.